zerver/lib/actions.py

```diff
@@ -501,11 +501,16 @@ def do_create_user(email: str, password: Optional[str], realm: Realm, full_name:
                    source_profile: Optional[UserProfile]=None,
                    realm_creation: bool=False) -> UserProfile:
 
+    # If a user is being created after authenticating using GitHub, than it doesn't make
+    # sense to have avatar_source == UserProfile.AVATAR_FROM_GRAVATAR and also
+    # user_avatar_url as their GitHub avatar url.
+    assert not (avatar_source == UserProfile.AVATAR_FROM_GRAVATAR and user_avatar_url)
```

Comment:
Same argument applies in `create_user`, so I think we can just move this assert into that function.